app/src/main/java/com/example/moviews/screen/home/adapter/ViewPagerAdapter.kt

```diff
@@ -0,0 +1,47 @@
+package com.example.moviews.screen.home.adapter
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.viewpager.widget.PagerAdapter
+import com.bumptech.glide.Glide
+import com.example.moviews.R
+import com.example.moviews.data.model.Movie
+import com.example.moviews.utils.Constant
+
+import kotlinx.android.synthetic.main.item_slide.view.*
+
+class ViewPagerAdapter(
+    private val onClickItem: (Movie) -> Unit
+) : PagerAdapter() {
+
+    private val movies = mutableListOf<Movie>()
+
+    override fun getCount(): Int = movies.size - 15
+
+    override fun instantiateItem(container: ViewGroup, position: Int): Any {
+        val view =
+            LayoutInflater.from(container.context).inflate(R.layout.item_slide, container, false)
+        Glide.with(view.context)
+            .load(Constant.BASE_URL_IMAGE + movies[position].backdrop)
+            .into(view.imageSlide)
+        view.textSlide.text = movies[position].title
+        view.setOnClickListener { onClickItem(movies[position]) }
+        container.addView(view, 0)
+        return view
+    }
+
+    override fun isViewFromObject(view: View, `object`: Any): Boolean = view == `object`
```

Comment:
đặt tên khác đi, hạn chế tên đặc biệt 
```suggestion
    override fun isViewFromObject(view: View, obj: Any): Boolean = view == `object`
```